java-checks/src/test/files/checks/security/TrustManagerCertificateCheck.java

```diff
@@ -0,0 +1,105 @@
+import javax.net.ssl.X509TrustManager;
+import java.security.cert.X509Certificate;
+import java.security.cert.CertificateException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class TrustAllManager implements X509TrustManager {
+
+  private static Logger LOG = Logger.getLogger("TrustAllManager");
+
+  @Override
+  public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException { // Noncompliant {{Change this method so it throws exceptions.}}
+  }
+
+  @Override
+  public void checkServerTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException { // Noncompliant
+    LOG.log(Level.SEVERE, "ERROR " + s);
+  }
+
+  @Override
+  public X509Certificate[] getAcceptedIssuers() {
+    return null;
+  }
+}
+
+class Main {
+
+  public static void main(String[] args) {
+
+    X509TrustManager trustManager = new X509TrustManager() {
+      @Override
+      public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException { // Noncompliant
+      }
+
+      @Override
+      public void checkServerTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException { // Noncompliant
+        System.out.println("123");
+      }
+
+      // does not override
+      public void checkServerTrusted(String s) {
+      }
+
+      @Override
+      public X509Certificate[] getAcceptedIssuers() {
+        return null;
+      }
+    };
+
+    trustManager = new X509TrustManager() {
+      @Override
+      public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
+        throw new CertificateException();
```

Comment:
Would it be worth to make sure it throws a CertificateException and not other types of exceptions?
So to add a test where it throws say IllegalArgumentException and make it NonCompliant?

Reply:
I was thinking about checking that the thrown exception is a CertificateException. But if you throw something else like:
throw new UnsupportedOperationException();
Then, there's no security issue.
An improvement could be to check that at least one of checkClientTrusted or checkServerTrusted throws CertificateException, but I think it does not worth the effort.